lib/charms/tls_certificates_interface/v4/tls_certificates.py

```diff
@@ -1057,12 +1057,28 @@ def _on_secret_expired(self, event: SecretExpiredEvent) -> None:
         try:
             csr_str = event.secret.get_content(refresh=True)["csr"]
         except ModelError:
-            logger.error("Failed to get CSR from secret - Skipping renewal")
+            logger.error("Failed to get CSR from secret - Skipping")
             return
         csr = CertificateSigningRequest.from_string(csr_str)
         self._renew_certificate_request(csr)
         event.secret.remove_all_revisions()
 
+    def renew_certificate(self, certificate: ProviderCertificate) -> None:
+        """Request the renewal of the provided certificate."""
+        certificate_signing_request = certificate.certificate_signing_request
+        secret_label = self._get_csr_secret_label(certificate_signing_request)
+        try:
+            secret = self.model.get_secret(label=secret_label)
+            current_csr = secret.get_content(refresh=True)["csr"]
+            if current_csr != str(certificate_signing_request):
+                logger.info("No matching CSR found - Skipping")
+                return
```

Comment:
I think this block does not have to be part of the try/except loop right? only `get_secret` can trigger SecretNotFound I believe

Reply:
Plus, in cases where we will not renew the certificate (ex. No matching CSR found or failed to get CSR from secret), we should log something higher than `info`. `warning` seems more appropriate.

Reply:
✅